### PR TITLE
VideoPlayer: Fix compiler warnings

### PR DESCRIPTION
**Pull request by garbear**
This PR fixes most compiler warnings due to type mismatches. Some type choices may need to be reevaluated.

### CODE REVIEW

File: xbmc/cores/VideoPlayer/VideoPlayer.cpp [hunk 1]
```diff
@@ -3590,15 +3590,17 @@ void CVideoPlayer::SetSpeed(float speed)
   if (!CanSeek())
     return;
   
-  m_newPlaySpeed = speed * DVD_PLAYSPEED_NORMAL;
+  int iSpeed = static_cast<int>(speed * DVD_PLAYSPEED_NORMAL); //! @todo
```
Comment (ksooo) [inline xbmc/cores/VideoPlayer/VideoPlayer.cpp]: @todo intended?

Comment (garbear) [inline xbmc/cores/VideoPlayer/VideoPlayer.cpp]: Hm, must have been unsure if an int was correct here

**Comment by fritsch** [comment]
@garbear as you see most of those methods and functions are touched in the vp_updates branch, which is on its way getting merged. Mind PRing this cleanup to this branch? That saves all that have PRs in the queue for the vp_updates branch a rebase?

**Comment by garbear** [comment]
I thought this pr was against `vpupdates`?

**Comment by fritsch** [comment]
@garbear perfect :-) sorry - blame it on the coffee. Thanks very much.

**Comment by Rechi** [comment]
@fritsch this PR is for vpupdates

**Comment by fritsch** [comment]
@Rechi jep - see above, realized my mystake. All fine.

File: xbmc/cores/VideoPlayer/VideoPlayer.cpp (hunk 1 above)
Comment (garbear) [inline xbmc/cores/VideoPlayer/VideoPlayer.cpp]: Oooh, I remember - I wasn't sure if I should cast or round. As it is, a speed of 0.98 or -0.98 will be converted to 0. We should probably just use a float throughout, which requires more extensive changes. Feedback?

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoPlayer.cpp]: DVD_PLAYSPEED_NORMAL is defiend as 1000. A speed of 0.9 resulted in new playspeed of 900. I want to keep this behaviour.

Comment (garbear) [inline xbmc/cores/VideoPlayer/VideoPlayer.cpp]: Ok, forgot about the dvd time base. This is OK then.

File: xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecFFmpeg.cpp [hunk 2]
```diff
@@ -151,8 +151,8 @@ bool CDVDAudioCodecFFmpeg::AddData(const DemuxPacket &packet)
   av_init_packet(&avpkt);
   avpkt.data = packet.pData;
   avpkt.size = packet.iSize;
-  avpkt.dts = (packet.dts == DVD_NOPTS_VALUE) ? AV_NOPTS_VALUE : packet.dts / DVD_TIME_BASE * AV_TIME_BASE;
-  avpkt.pts = (packet.pts == DVD_NOPTS_VALUE) ? AV_NOPTS_VALUE : packet.pts / DVD_TIME_BASE * AV_TIME_BASE;
+  avpkt.dts = (packet.dts == DVD_NOPTS_VALUE) ? AV_NOPTS_VALUE : static_cast<int64_t>(packet.dts / DVD_TIME_BASE) * AV_TIME_BASE;
+  avpkt.pts = (packet.pts == DVD_NOPTS_VALUE) ? AV_NOPTS_VALUE : static_cast<int64_t>(packet.pts / DVD_TIME_BASE) * AV_TIME_BASE;
```
Comment (garbear) [inline xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecFFmpeg.cpp]: are these casts correct?

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecFFmpeg.cpp]: I don't think so. With this cast you only get multiples of AV_TIME_BASE.
Did you runtime test this? :)

Comment (garbear) [inline xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecFFmpeg.cpp]: yea... but I thought the stuttering was due to some VP work I had on top of this :) will PR a fix

**Comment by Rechi** [comment]
@garbear @FernetMenta this causes grave video stuttering

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp [hunk 3]
```diff
@@ -479,8 +479,8 @@ bool CDVDVideoCodecFFmpeg::AddData(const DemuxPacket &packet)
   av_init_packet(&avpkt);
   avpkt.data = packet.pData;
   avpkt.size = packet.iSize;
-  avpkt.dts = (packet.dts == DVD_NOPTS_VALUE) ? AV_NOPTS_VALUE : packet.dts / DVD_TIME_BASE * AV_TIME_BASE;
-  avpkt.pts = (packet.pts == DVD_NOPTS_VALUE) ? AV_NOPTS_VALUE : packet.pts / DVD_TIME_BASE * AV_TIME_BASE;
+  avpkt.dts = (packet.dts == DVD_NOPTS_VALUE) ? AV_NOPTS_VALUE : static_cast<int64_t>(packet.dts / DVD_TIME_BASE) * AV_TIME_BASE;
+  avpkt.pts = (packet.pts == DVD_NOPTS_VALUE) ? AV_NOPTS_VALUE : static_cast<int64_t>(packet.pts / DVD_TIME_BASE) * AV_TIME_BASE;
```
Comment (Rechi) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp]: wrong casts
they cause the video stuttering I mentioned

**Pull request merged**